akka-docs/rst/java/code/docs/persistence/PersistentActorExample.java

```diff
@@ -97,16 +98,14 @@ public Receive createReceive() {
         return receiveBuilder()
             .match(Cmd.class, c -> {
               final String data = c.getData();
-              final Evt evt1 = new Evt(data + "-" + getNumEvents());
-              final Evt evt2 = new Evt(data + "-" + (getNumEvents() + 1));
-              persistAll(asList(evt1, evt2), (Evt evt) -> {
-                state.update(evt);
-                if (evt.equals(evt2)) {
-                  getContext().system().eventStream().publish(evt);
-                }
+              final Evt evt = new Evt(data + "-" + getNumEvents());
+              persist(evt, (Evt e) -> {
+                  state.update(e);
+                  getContext().system().eventStream().publish(e);
+                  if (lastSequenceNr() % snapShotInterval == 0 && lastSequenceNr() != 0)
+                      saveSnapshot(state);
```

Comment:
// IMPORTANT: create a copy of snapshot because ExampleState is mutable
    saveSnapshot(state.copy());